### PR TITLE
Patch codecoverage

### PR DESCRIPTION
**Pull request by choelzl**

### CODE REVIEW

**Review by SachaKor — APPROVED** [review]

**Comment by choelzl** [comment]
@PeterKrcmar0  I tried using your repository as inspiration and/or adding more paths but it still doesnt uses non-android unitTests for code-coverage.

EDIT: closed the PR for now

**Review by Yabdro — APPROVED** [review]

File: app/build.gradle [hunk 1]
```diff
@@ -103,13 +103,13 @@ task jacocoTestReport(type: JacocoReport, dependsOn: ['testDebugUnitTest', 'crea
...
-    def debugTree = fileTree(dir: "$project.buildDir/intermediates/classes/debug", excludes: fileFilter)
+    def fileFilter = ['**/R.class', '**/R$*.class', '**/BuildConfig.*', '**/Manifest*.*', '**/*Test*.*', 'test/**/*.*', '**/*test*.*', 'android/**/*.*']
+    def debugTree = fileTree(dir: "$project.buildDir/intermediates/javac/debug/classes", excludes: fileFilter)
     def mainSrc = "$project.projectDir/src/main/java"
 
     sourceDirectories.from = [mainSrc]
     classDirectories.from = [debugTree]
     executionData.setFrom(project.fileTree(dir: project.buildDir, includes: [
-            'jacoco/testDebugUnitTest.exec', 'outputs/code-coverage/connected/*coverage.ec'
+            'jacoco/testDebugUnitTest.exec', 'outputs/code-coverage/connected/*coverage.ec', 'outputs/code_coverage/debugAndroidTest/connected/*coverage.ec'
```
Comment (PeterKrcmar0) [inline app/build.gradle:113]: You can remove the middle one, since this file doesn't exist it could mess things up.

**Comment by PeterKrcmar0** [comment]
okkk couple of things to look at:

- if you change the jacoco version to 0.8.5 in app/build.gradle, make sure to also change it in the top level build.gradle
- in android studio, clean your project and then in the android studio terminal, run `gradlew build connectedCheck jacocoTestReport`. Then open up the `app/build/reports/jacoco/jacocoTestReport/html/index.html` file in a browser to check if any code was covered
- if not, then maybe your tests just aren't run at all. try adding `--info` at the end of the command and check the output to see if your tests are actually executed (try to find the names of your tests i.e. constructor(), getOwner(), ...)

File: app/build.gradle (hunk 1 above)
Comment (PeterKrcmar0) [reply app/build.gradle:113]: I also see that in dependencies you have: `testImplementation "org.junit.jupiter:junit-jupiter-api:5.3.2"`
Try changing that to `testImplementation 'junit:junit:4.12'`

Comment (choelzl) [reply app/build.gradle:113]: We wanted to take advantage of some features from jupiter/junit5 (that don't exist in junit4)

**Comment by PeterKrcmar0** [comment]
**UPDATE:**
I looked at it locally and found that the issue was indeed caused by junit5. You can either switch back to junit4 and slightly adapt your tests or try to configure jacoco with junit5 (I did some very quick research and it doesn't seem trivial). I advise you stick with junit4 for now, as IMO it:
- offers enough functionality
- will spare you a lot of time

To adapt your tests, you need to:
- set the class to `public`
- set all unit tests to `public`
- replace `@BeforeEach` with `@Before`
- maybe some other minor things

As a side note, I had some failing tests on my computer so make sure that they all pass !

**Pull request closed without merging**